src/test/ui/async-await/async-fn-multiple-lifetimes.rs

```diff
@@ -1,11 +1,13 @@
 // edition:2018
+// run-pass
 
-#![feature(arbitrary_self_types, async_await, await_macro, pin)]
+// Test that we can use async fns with multiple arbitrary lifetimes.
```

Comment:
Would be good to have more tests for the stable surface?

Reply:
Especially for `async fn() -> impl Trait`

Reply:
It would, yes =)

Reply:
> Especially for `async fn() -> impl Trait`

Ah, interesting. I think that should work, but that's a good corner case to cover!

Reply:
Added more tests around this. I'm going to mark this concern as resolved for now.